### PR TITLE
WESTPA2-threshold2 fix

### PR DESCRIPTION
**Pull request by jeremyleung521**
TL:DR - Made changes. Not heavily tested at all so don't merge it right off the bat. Also added a warning if min_weight < 1e-100.

Just documenting my changes/rational:
I added in an extra condition to each of the `while` loops in `adjust_counts` so they can run a number of times until the "bin" object is unchanges (ie. list of segments from the previous for loop == the next one). This is to prevent the code from exiting prematurely just because the `for` loop ran once (i.e. the `elif` statement conditions were initially too broad). This change should not affect the threshold usage at all and now allows the driver to reproduce intended behavior from the original code. 

The `test_recycle` test in `test_we_driver.py` actually requires the split `for` loop in `adjust_counts` to execute twice in order to get the final 4 segments. The changes in this branch only allow the for loop to run exactly once (and thus breaking out of the `while` loop even when a threshold is not explicitly specified), so the test fails. Actually glad the CI caught it.

### CODE REVIEW

**Comment by AnthonyBogetti** [comment]
I tested this and it works; it also passes the pytest for the we_driver.
